File: 2020/5.kt
// Advent of code 2020 day 5
import java.io.File

fun main() {
  val fileName = "input5.txt"
  val lines: List<String> = File(fileName).readLines()

  val rows: MutableMap<Int, MutableList<Int>> = mutableMapOf<Int, MutableList<Int>>()

  for (line in lines) {
    var row = Integer.parseInt(line.replace("B", "1").replace("F", "0").substring(0, 7), 2)
    var seat = Integer.parseInt(line.replace("R", "1").replace("L", "0").takeLast(3), 2)

    if (rows.containsKey(row)) {
      rows[row]?.add(seat)
    } else {
      rows[row] = mutableListOf<Int>(seat)
    }
  }

  for ((row, seats) in rows.toSortedMap()) {
    //println("${row.toString().padStart(5)}, ${seats.sorted()}")
    if (seats.size != 7) {
      continue
    }
    for (i in 0..7) {
      if (seats.contains(i) == false) {
        val value = 8 * row + i
        println( "$value")
      }
    }
  }
}

File: 2020/5.py
#!/usr/bin/env python

inputfile = "input5.txt"
with open(inputfile, "r") as f:
  lines = [x.strip() for x in f.readlines()]

rows = {}

for line in lines:
  row = line.replace("B", "1")[0:7]
  row = int(row.replace("F", "0"), 2)

  seat = line.replace("L", "0")[7:]
  seat = int(seat.replace("R", "1"), 2)

  #print row, seat, row*8 + seat

  try:
    rows[row].append(seat)
  except KeyError:
    rows[row] = [seat]

for row in rows:
  print str(row).zfill(3), sorted(rows[row])
